File: MySQL/Python/Select/mysql_select.py
from __future__ import print_function

import mysql.connector
from mysql.connector import errorcode

def select():
    try:
        print("Connecting to MySQL database...")
        cnx = mysql.connector.connect(user='root', password='My$ql@Server#5.7',
                                  host='localhost', database='classicmodels')

        if cnx.is_connected():
            print("...connection established.")
        else:
            print("...connection failed.")
            
        cursor = cnx.cursor()
        # query = "SELECT * FROM customers ORDER BY customerNumber DESC LIMIT 2;"
        query = "SELECT * FROM customers;"
        cursor.execute(query)

        # for record in cursor:
            # print(record)

        rows = cursor.fetchall()
        print('Number of Row(s) =', cursor.rowcount)
        
        for row in rows:
            # print(row)
            counter = 0
            for column in row:
                print(column, end="")
                counter = counter + 1
                if counter < len(row):
                    print(" - ", end="")                
            print("")
    except mysql.connector.Error as err:
        if err.errno == errorcode.ER_ACCESS_DENIED_ERROR:
            print("Something is wrong with your user name or password")
        elif err.errno == errorcode.ER_BAD_DB_ERROR:
            print("Database does not exist")
        else:
            print(err)
    finally:
        cursor.close()
        cnx.close()
        print("Connection closed.")

if __name__ == "__main__":
    select()

File: MySQL-MariaDB/Node.js/Select/mysql_select.js
var mysql      = require('mysql');
var connection = mysql.createConnection({
  host     : 'localhost',
  user     : 'root',
  password : 'My$ql@Server#5.7',
  database : 'classicmodels',
});

// var post  = {id: 1, title: 'Hello MySQL'};
var post  = ""

connection.connect(function(err) {
  // connected! (unless `err` is set)
  console.log("...connection established.")
  // var sql = "SELECT * FROM customers ORDER BY customerNumber DESC LIMIT 2;"
  var sql = "SELECT * FROM customers;"
  var query = connection.query(sql, post, function(err, results) {
    //console.log(results);
    console.log(results.length);
    var numCallbacks = 0;
    for (var index in results){
      // console.log(results[index]);
      // console.log(JSON.stringify(results[index]));
      var stringified = JSON.stringify(results[index]);
      var parsed = JSON.parse(stringified);
      console.log(parsed.customerNumber + " - " + parsed.customerName + " - " + parsed.contactLastName + " - " + parsed.contactFirstName + " - " + 
                  parsed.phone + " - " + parsed.addressLine1 + " - " + parsed.addressLine2 + " - " + parsed.city + " - " + parsed.state + " - " + 
                  parsed.postalCode + " - " + parsed.country + " - " + parsed.salesRepEmployeeNumber + " - " + parsed.creditLimit);
      if (++numCallbacks == results.length) {
        console.log("Connection end.")
        return connection.end();
      }
    }    
  });
  //console.log(query.sql);
});
